components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -1260,6 +1252,20 @@ impl Animatable for LengthOrPercentageOrAuto {
     }
 }
 
+impl ToAnimatedZero for LengthOrPercentageOrAuto {
+    #[inline]
+    fn to_animated_zero(&self) -> Result<Self, ()> {
+        match *self {
+            LengthOrPercentageOrAuto::Length(_) |
+            LengthOrPercentageOrAuto::Percentage(_) |
+            LengthOrPercentageOrAuto::Calc(_) => {
+                Ok(LengthOrPercentageOrAuto::Length(Au(0)))
```

Comment:
I guess this one seems somewhat weird, in the sense of it converts `foo%` into `0px`, but since it's preserving it it's probably not a big deal.